provider/datagen/src/cldr/transform/datetime/mod.rs

```diff
@@ -149,8 +160,8 @@ mod test {
     #[test]
     fn test_basic_patterns() {
         let cldr_paths = crate::cldr::cldr_paths::for_test();
-        let provider = CommonDateProvider::try_from(&cldr_paths as &dyn CldrPaths)
-            .expect("Failed to retrieve provider");
+        let provider =
+            CommonDateProvider::try_new(&cldr_paths).expect("Failed to retrieve provider");
```

Comment:
Thought: provide a `DatagenOptions::for_test` to keep tests consistent with what's happening irl.

Reply:
Nit: This comment looks like a Suggestion, not a Thought. A Thought is "non-blocking by nature."

https://conventionalcomments.org/

Reply:
Okay, I added `DatagenOptions::for_test()`, but in order to do this, I had to convert it to be fully owned, since the testdata paths are owned. I could have done Cow but I don't want to overcomplicate things.

I'm not currently using `DatagenOptions::for_test()` anywhere because I did not write these `From` impls with the idea that they would be for general use at time time. The `try_new` functions are the ones to use for creating specific instances of transformers.

Reply:
yeah sorry these are suggestions.

Why keep the `try_new`s though? If we can instantiate each transformer from the same struct I think it would simplify things because we wouldn't have to special-case between CLDR, uprops, segmenter, tzif, etc. It would also reduce boilerplate for transformers which use multiple data-sources, like list and segmenter.

Reply:
For now, I see try_new as low-level constructors, which may be useful if you're creating only one specific type of transformer, and the new From impl as a higher level one

Reply:
For now, I see try_new as low-level constructors, which may be useful if you're creating only one specific type of transformer, and the new From impl as a higher level one. If we do your other proposal about error messages, these constructors are going to need to change again anyway...